### ISSUE TITLE
Bug report: user profile not appearing in profile relationship dropdown.

### ISSUE THREAD

**Issue by theMackabu** [pocketbase/pocketbase#238]
To get expand working on a user profile, it must be assigned as a relation, but I cant seem to assign users through the admin panel
![Screen Shot 2022-07-29 at 2 50 31 PM](https://user-images.githubusercontent.com/32078755/181739356-3521701b-98b3-48c9-96e2-5d4b4b43673e.png)
![Screen Shot 2022-07-29 at 2 50 38 PM](https://user-images.githubusercontent.com/32078755/181739364-4bcf4e50-66c3-4b39-8632-c7a4829e2513.png)
![Screen Shot 2022-07-29 at 2 50 45 PM](https://user-images.githubusercontent.com/32078755/181739377-bb3e135c-32b4-40dc-82d7-b25cde812fb3.png)
Screenshots attached above

**Comment by ganigeorgiev** [comment]
Hi,
I've just tested it locally and I cannot reproduce the issue - the profiles records are visible and I'm able to select them.

Are you able to reproduce the issue on the live demo - https://pocketbase.io/demo/ (v0.3.2)?

**Comment by theMackabu** [comment]
I cannot reproduce the issue on the live demo, and the request that involve `/api/collections` are 
![Screen Shot 2022-07-29 at 9 36 17 PM](https://user-images.githubusercontent.com/32078755/181809526-7405a349-e737-49ef-b2dd-d499d1446fc7.png)

when clicking on the item and opening the dropdown, this request is sent
![Screen Shot 2022-07-29 at 9 36 26 PM](https://user-images.githubusercontent.com/32078755/181809566-0f3479be-ebc3-4875-b453-261aeab42370.png)
but the dropdown is still empty

**Comment by theMackabu** [comment]
This couldn't possibly break it right?

Original code for user api
![Screen Shot 2022-07-29 at 9 53 15 PM](https://user-images.githubusercontent.com/32078755/181812075-86e5284e-9084-48b4-93cc-545c3d2614f9.png)

Modified user api
![Screen Shot 2022-07-29 at 9 53 20 PM](https://user-images.githubusercontent.com/32078755/181812091-93351aa6-d308-4340-ac28-3d3bfb19a6e2.png)

**Comment by theMackabu** [comment]
> Changing the middleware for the user update action shouldn't be the cause for the missing dropdown items, but modifying the source is not the correct way to extend PocketBase. You should use only the hooks or create a custom action, otherwise you'll have to keep your project in sync manually with new updates.
> 
> On what PocketBase your changes are based? In v0.2.8 a fix was added to fix a similar issue with the user/relation fields - [#100 (comment)](https://github.com/pocketbase/pocketbase/issues/100#issuecomment-1185729831).
> 
> Have you made any other changes to the original API?
> 
> Are there any error logs in the browsers Inspect tool console?

No errors in the console, the only other change is the the embeded spa, and the version is based on the lastest pull from the repo

**Comment by ganigeorgiev** [comment]
Could you try downloading the latest binary for your platform - https://github.com/pocketbase/pocketbase/releases/tag/v0.3.2
and start the executable inside it by pointing to your current `pb_data` directory, eg:
```
./pocketbase --dir="/path/to/your/pb_data"
```

If the admin that comes with the above executable shows the profiles correctly, then there is something in your changes that breaks the dropdown for some reason but I don't know how to debug or reproduce it (please make sure to check again that there are no js error/warning in the browser's console).

**Comment by theMackabu** [comment]
> Could you try downloading the latest binary for your platform - https://github.com/pocketbase/pocketbase/releases/tag/v0.3.2 and start the executable inside it by pointing to your current `pb_data` directory, eg:
> 
> ```
> ./pocketbase --dir="/path/to/your/pb_data"
> ```
> 
> If the admin that comes with the above executable shows the profiles correctly, then there is something in your changes that breaks the dropdown for some reason but I don't know how to debug or reproduce it (please make sure to check again that there are no js error/warning in the browser's console).

I will try that now, but the console is only just a few react dev tool warnings and slow internet warnings
![Screen Shot 2022-07-29 at 10 46 12 PM](https://user-images.githubusercontent.com/32078755/181820219-41185c94-c049-42af-ac4d-8c7ab8b01367.png)

**Comment by ganigeorgiev** [comment]
I'm really confused.

Have you done any changes in the Admin UI? I cannot find the `index.78281687.js` from the above logs in the latest `ui/dist` (or in the last couple previous releases), which implies that you've manually built the ui, right?

What is loading the `inject.js.map` that results in 404; is there a `inject.js` script somewhere or is this from some browser extension?

**Comment by theMackabu** [comment]
Inject.js is from some extension that broke and the dev has not updated it yet, but disabling it gives the same result. Admin UI was rebuilt because of github actions (thats how I get my binary)

**Comment by theMackabu** [comment]
Just ran a clean download from github, and ran the pocketbase, still results in 
![Screen Shot 2022-07-29 at 11 39 07 PM](https://user-images.githubusercontent.com/32078755/181828239-092e27c9-1889-44ca-9259-0e49c5bd43c0.png)

**Comment by ganigeorgiev** [comment]
Sorry, I don't know how to reproduce it.

A couple of things to try:
- Open the admin in Incognito/Private mode in order to rule out that there is an extension that interferes with the UI.
- Make sure `profile` relation fields is actually pointing to the user profiles collection
- Try again with the latest release binary but this time with a clean `pb_data` and try to reproduce the issue

**Comment by theMackabu** [comment]
Tested in Chrome(android, incognito mode)
![Screenshot_20220730-000659_Chrome](https://user-images.githubusercontent.com/32078755/181832020-880fdb5c-1f86-4ae4-bff9-de8cce26264d.png)
![Screenshot_20220730-000647_Chrome](https://user-images.githubusercontent.com/32078755/181832030-36a32b95-f319-47ee-aefe-5401839e83d2.png)
![Screenshot_20220730-000637_Chrome](https://user-images.githubusercontent.com/32078755/181832034-ffe9fdaf-422a-4023-a47c-d5947bb40abc.png)

**Comment by theMackabu** [comment]
Testing if pb_data changes it, will update this comment accordingly

**Comment by ganigeorgiev** [comment]
You could try running a `SELECT * FROM _collections` query via the sqlite3 cli against the "broken" db and post it here, eg.
```
sqlite3 pb_data/data.db "select * from _collections;"
```

**Comment by theMackabu** [comment]
> You could try running a `SELECT * FROM _collections` query via the sqlite3 cli against the "broken" db and post it here, eg.
> 
> ```
> sqlite3 pb_data/data.db "select * from _collections"
> ```

```
60JgfaEFt44JljE|1|profiles|[{"system":true,"id":"ufalimoz","name":"userId","type":"user","required":true,"unique":true,"options":{"maxSelect":1,"cascadeDelete":true}},{"system":false,"id":"znfrpszc","name":"username","type":"text","required":true,"unique":false,"options":{"min":null,"max":null,"pattern":""}},{"system":false,"id":"fmdt9v78","name":"avatar","type":"file","required":false,"unique":false,"options":{"maxSelect":1,"maxSize":5242880,"mimeTypes":["image/jpg","image/jpeg","image/png","image/svg+xml","image/gif"],"thumbs":null}},{"system":false,"id":"6u0qlcbr","name":"about","type":"text","required":false,"unique":false,"options":{"min":null,"max":null,"pattern":""}},{"system":false,"id":"d1nbswv6","name":"banner","type":"text","required":false,"unique":false,"options":{"min":null,"max":null,"pattern":""}},{"system":false,"id":"ezt3a4id","name":"firstname","type":"text","required":true,"unique":false,"options":{"min":null,"max":null,"pattern":""}},{"system":false,"id":"ctc1d5sz","name":"lastname","type":"text","required":true,"unique":false,"options":{"min":null,"max":null,"pattern":""}},{"system":false,"id":"k2d92izy","name":"group","type":"select","required":true,"unique":false,"options":{"maxSelect":1,"values":["admin","seller","user"]}},{"system":false,"id":"r2vdz2um","name":"friends","type":"user","required":false,"unique":false,"options":{"maxSelect":100,"cascadeDelete":false}},{"system":false,"id":"sounff4z","name":"sales","type":"number","required":false,"unique":false,"options":{"min":0,"max":null}},{"system":false,"id":"f7lfw0om","name":"discord","type":"text","required":true,"unique":true,"options":{"min":null,"max":null,"pattern":"^.{3,32}#[0-9]{4}$"}},{"system":false,"id":"qkutf7cb","name":"publicEmail","type":"text","required":true,"unique":false,"options":{"min":null,"max":null,"pattern":""}},{"system":false,"id":"tmx1wbsg","name":"seller_status","type":"select","required":false,"unique":false,"options":{"maxSelect":1,"values":["normal","verified"]}}]||||userId = @request.user.id||2022-07-25 10:09:45.588|2022-07-27 10:30:32.852
i4rFwvg2DyA2muh|0|resources|[{"system":false,"id":"toyspaqr","name":"name","type":"text","required":true,"unique":false,"options":{"min":null,"max":null,"pattern":""}},{"system":false,"id":"b6ill2rt","name":"price","type":"number","required":true,"unique":false,"options":{"min":null,"max":null}},{"system":false,"id":"vqg4l8qx","name":"icon","type":"file","required":true,"unique":false,"options":{"maxSelect":1,"maxSize":5242880,"mimeTypes":[],"thumbs":[]}},{"system":false,"id":"rfyvf978","name":"banner","type":"file","required":false,"unique":false,"options":{"maxSelect":1,"maxSize":5242880,"mimeTypes":[],"thumbs":[]}},{"system":false,"id":"wdrrzguf","name":"description","type":"text","required":false,"unique":false,"options":{"min":null,"max":null,"pattern":""}},{"system":false,"id":"mvuqs2bq","name":"content","type":"text","required":false,"unique":false,"options":{"min":null,"max":null,"pattern":""}},{"system":false,"id":"amnprmqc","name":"file","type":"relation","required":false,"unique":false,"options":{"maxSelect":20,"collectionId":"9tIzsBXLcEF66rM","cascadeDelete":false}},{"system":false,"id":"hehzcon9","name":"category","type":"select","required":true,"unique":false,"options":{"maxSelect":1,"values":["addons","eggs","scripts","services","plugins","themes","other"]}},{"system":false,"id":"g4xnerzz","name":"author","type":"user","required":true,"unique":false,"options":{"maxSelect":1,"cascadeDelete":false}},{"system":false,"id":"chokiklw","name":"profile","type":"relation","required":true,"unique":false,"options":{"maxSelect":1,"collectionId":"60JgfaEFt44JljE","cascadeDelete":false}}]||||||2022-07-25 11:24:58.643|2022-07-29 10:20:22.837
9tIzsBXLcEF66rM|0|files|[{"system":false,"id":"gwgwlraq","name":"filename","type":"text","required":true,"unique":false,"options":{"min":null,"max":null,"pattern":""}},{"system":false,"id":"sjfa7lb5","name":"size","type":"number","required":true,"unique":false,"options":{"min":null,"max":null}},{"system":false,"id":"fassaige","name":"type","type":"text","required":true,"unique":false,"options":{"min":null,"max":null,"pattern":""}},{"system":false,"id":"pjisgbqt","name":"data","type":"file","required":true,"unique":false,"options":{"maxSelect":1,"maxSize":5242880,"mimeTypes":[],"thumbs":[]}},{"system":false,"id":"hzyau4ci","name":"uploader","type":"user","required":true,"unique":false,"options":{"maxSelect":1,"cascadeDelete":false}}]||||||2022-07-25 11:27:02.835|2022-07-25 11:27:02.835
```

**Comment by ganigeorgiev** [comment]
The output seems OK - the `resources` are correctly wired to the `profiles` collection.

There seems to be something else going on but I can't think of anything else that could break it. If you have at least one profile record, then it should be visible in the relation field dropdown (you could try with creating another relation field pointing to a different collection).

I've tried it with several different configurations (including the schemas from above), but I'm not able to reproduce it.

@theMackabu Unless you are able to share a minimal reproducible example, I'm not sure that I can help you, so I'll close the issue for now as "Cannot reproduce".

**Comment by theMackabu** [comment]
> The output seems OK - the `resources` are correctly wired to the `profiles` collection.
> 
> There seems to be something else going on but I can't think of anything else that could break it. If you have at least one profile record, then it should be visible in the relation field dropdown (you could try with creating another relation field pointing to a different collection).
> 
> I've tried it with several different configurations (including the schemas from above), but I'm not able to reproduce it.
> 
> @theMackabu Unless you are able to share a minimal reproducible example, I'm not sure that I can help you, so I'll close the issue for now as "Cannot reproduce".

![Screen Shot 2022-07-31 at 7 06 27 PM](https://user-images.githubusercontent.com/32078755/182031428-74a9b2a1-19fa-46b8-9dd0-9c0d96abdc91.png)
other relations work fine, just profiles arent. Is it possible to nuke/wipe the profiles collection or transfer the two other collections to a fresh db?

**Comment by theMackabu** [comment]
I found how to reproduce it
I have created a friends relation with 100 or more user relations, then it will not appear in the menu

**Comment by theMackabu** [comment]
attached zip with demo database reproduced below
[pb_data.zip](https://github.com/pocketbase/pocketbase/files/9229099/pb_data.zip)

**Comment by ganigeorgiev** [comment]
@theMackabu Thanks! I've found where the issue is and why I couldn't reproduce it previously. The reason for the bug is the `group` named field, because in the code when populating the select we are expecting the select option items to be either a flatten list with items or an array with the following structure `[{group: 'My group', items: [...]}]` and since the `group` key collides and ended up in the second case where the profile response doesn't have an `items` array and it doesn't show anything.

The code responsible for this is here - https://github.com/pocketbase/pocketbase/blob/master/ui/src/components/base/Select.svelte#L13-L33, but I'll have to think a little more how to fix it (one way will be to remove the implicit group conversion but I'll have to test it further and will let you know when I push a fix in master).

**Comment by theMackabu** [comment]
> @theMackabu Thanks! I've found where the issue is and why I couldn't reproduce it previously. The reason for the bug is the `group` named field, because in the code when populating the select we are expecting the select option items to be either a flatten list with items or an array with the following structure `[{group: 'My group', items: [...]}]` and since the `group` key collides and ended up in the second case where the profile response doesn't have an `items` array and it doesn't show anything.
> 
> The code responsible for this is here - https://github.com/pocketbase/pocketbase/blob/master/ui/src/components/base/Select.svelte#L13-L33, but I'll have to think a little more how to fix it (one way will be to remove the implicit group conversion but I'll have to test it further and will let you know when I push a fix in master).

So it wasn't thr friends field,, good to know, and that makes sense

**Comment by ganigeorgiev** [comment]
@theMackabu I've removed the implicit select items grouping - we are not using it anywhere and it just add unnecessary complexity to the reactivity of the select component.

The fix currently is only in the master branch, so you can pull the changes from there if  you want to test them.

I'm planning a v0.4.0 release sometime next week with other minor bugfixes (including query routing rewrite because of #204) and small improvements related to the realtime subscriptions.

Thanks again for isolating the bug!

**Comment by theMackabu** [comment]
Alright, thanks so much

**Comment by ganigeorgiev** [comment]
@theMackabu Just to let you know that the fixes is also part of [v0.3.4 release](https://github.com/pocketbase/pocketbase/releases/tag/v0.3.4).
